connector/web3/web3_test.go

```diff
@@ -0,0 +1,143 @@
+package web3
+
+import (
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"github.com/dexidp/dex/connector"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+
+	"log"
+	"testing"
+)
+
+func newConnector(t *testing.T) *web3Connector {
+	log := logrus.New()
+
+	testConfig := Config{
+		InfuraID: "mockInfuraID",
+	}
+
+	conn, err := testConfig.Open("id", log)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	web3Conn, ok := conn.(*web3Connector)
+	if !ok {
+		t.Fatal(err)
+	}
+
+	return web3Conn
+}
+
+func generateWallet() (*ecdsa.PrivateKey, *common.Address, error) {
+	privateKey, err := crypto.GenerateKey()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	userAddr := crypto.PubkeyToAddress(privateKey.PublicKey)
+
+	return privateKey, &userAddr, nil
+}
+
+func signMessage(msg string, pk *ecdsa.PrivateKey) ([]byte, []byte) {
+	data := []byte(msg)
+	hash := accounts.TextHash(data)
+
+	signature, err := crypto.Sign(hash, pk)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return signature, hash
+}
+
+func TestEOALogin(t *testing.T) {
```

Comment:
Test cases are cool. Could we include one that has the v param as 27 or 28? Then we could test that subtraction on line 48 of `web3.go`.